sql/expression/convert.go

```diff
@@ -240,3 +246,18 @@ func convertValue(val interface{}, castTo string, originType sql.Type) (interfac
 		return nil, nil
 	}
 }
+
+// convertHexBlobToDecimalForNumericContext returns numeric value if the value is binary and origin type is blob
+// IF the convertTo type is number type. This applies only for hex LITERAL values parsed it is parsed as
+// BLOB type in the parser but to binary string type.
+// -- other byte arrays of other sql types does not apply here.
+func convertHexBlobToDecimalForNumericContext(val interface{}, originType sql.Type) interface{} {
+	if bin, isBinary := val.([]byte); isBinary && sql.IsBlobType(originType) {
+		stringVal := hex.EncodeToString(bin)
+		decimalNum, err := strconv.ParseUint(stringVal, 16, 64)
```

Comment:
Would be good to do something with this `err` so we don't swallow any errors and make future debugging harder. Think this should error the statement if this happens?